app/models/recruitment_cycle.rb

```diff
@@ -5,21 +5,25 @@ module RecruitmentCycle
   }.freeze
 
   def self.current_year
-    if Time.zone.today < EndOfCycleTimetable.find_reopens
-      2020
-    else
-      2021
+    cycle = EndOfCycleTimetable::CYCLE_DATES.detect do |year, dates|
+      now = Time.zone.today
+      find_opens = dates[:find_reopens]
+      find_reopens = EndOfCycleTimetable::CYCLE_DATES[year + 1][:find_reopens]
+
+      now.between?(find_opens, find_reopens)
     end
-  end
 
-  def self.previous_year
-    current_year - 1
+    cycle[0]
```

Comment:
`cycle` returns an array with `[year, cycle_dates]` - is there a nicer way to do this? Possibly we may want to change how the cycle dates are structured further down the line?

Reply:
One of the tasks in the refactoring plan is to merge the RecruitmentCycle class and EndOfCycleTimetable class, however we think there may be a stronger case for keeping them separate as these are distinct responsibilities. What do other people think?

Reply:
I think moving implementation into `EndOfCycleTimetable` better insulates us from future changes to the way we structure cycle dates.

Maybe we could iterate over keys like this? (untested 😬)

```ruby
now = Time.zone.today
CYCLE_DATES.keys.detect do |year|
  now.between(
    CYCLE_DATES[year][:find_reopens],
    CYCLE_DATES[year + 1][:find_reopens],
  )
end
```

I don't have strong opinions on whether we have to merge these things — it's definitely useful to be able to consider cycles at the "year" resolution! Could probably offer that equally well as methods on the timetable or as now in a separate class ?

Reply:
Yep that code looks good. Have updated!